build/cmake/CMakeLists.txt

```diff
@@ -11,6 +11,8 @@ set (build_cmake_MODULES
    MongoCPackage.cmake
    ParseVersion.cmake
    SphinxBuild.cmake
+   CCache.cmake
+   LLDLinker.cmake
```

Comment:
I suggest renaming `CCache.cmake` and `LLDLinker.cmake` to `FindCcache.cmake` and `FindLLD.cmake` respectively, following the convention used by `FindSnappy.cmake`, `FindSASL2.cmake`, etc.

Reply:
I did not mean to suggest supporting `find_package()` usage. This was just about filename consistency. For example, `FindSnappy.cmake` is included as [`include (FindSnappy)`](https://github.com/mongodb/mongo-c-driver/blob/932e7e4861e28a719969cf716cdc4c10473e78ee/src/libmongoc/CMakeLists.txt#L353), and `FindSASL2` is included as [`include (FindSASL2)`](https://github.com/mongodb/mongo-c-driver/blob/932e7e4861e28a719969cf716cdc4c10473e78ee/src/libmongoc/CMakeLists.txt#L252); neither are used with, nor appear to designed to be used with, `find_package()`.

Reply:
CMake modules with the magic form `FindFoo` are intended to be used with `find_package(Foo)`, which makes `include(FindSnappy)` and `include(FindSASL2)` to be "wrong", even if they incidentally "just work" as they currently are. It's likely an accident that they were given that name and then used with `include()`. Those should probably be patched up to use `find_package_handle_standard_args()` (and maybe imported libraries as well).

Reply:
Fair points. Agreed that `Find<Name>.cmake` files are usually reserved for `find_package()` usage.